galore/cross_sections.py

```diff
@@ -281,3 +281,261 @@ def _eval_fit(energy, coeffs):
+    electrons_numbers_arrays = np.array_split(electrons_number, 2)
+
+    subshells_cross_sections_sum = np.array(
+        list(map(sum, subshells_cross_section_arrays)))
+
+    subshells_electrons_number_sum = np.array(
+        list(map(sum, electrons_numbers_arrays)))
+
+    avg_orbital_cross_sections = np.true_divide(
+        subshells_cross_sections_sum, subshells_electrons_number_sum)
+
+    return avg_orbital_cross_sections
+
+
+def _cross_sections_from_csv_data(energy, data, reference):
+    """
+    Get cross sections from data dict 
+    For known sources, data is based on tabulation of Yeh/Lindau (1985).[1]
+    Otherwise, energies in keV from 1-1500 are used with log-log polynomial
+    parametrisation of data from Scofield.[2]
+
+    References:
+      1. Yeh, J.J. and Lindau, I. (1985)
+         Atomic Data and Nuclear Data Tables 32 pp 1-155
+      2. J. H. Scofield (1973) Lawrence Livermore National Laboratory
+         Report No. UCRL-51326
+
+    Args:
+        energy(float): energy value  
+        data(dict): data from read_csv_file()
+        reference(str): 'Scofield' or 'Yeh'
+
+    Returns:
+        orbitals_cross_sections_dict: containing orbitals 's', 'p', 'd', 'f' and 
+                                      maximum cross sections of each orbital. 
+                                      Missing data is represented as None.
+
+    """
+
+    # replace '' in data table with NaN
+    for row in range(len(data['data_table'])):
+        data['data_table'][row] = [
+            float('NaN') if x == '' else x for x in data['data_table'][row]]
+
+    # change the data_table and electron_counts to float arrays
+    data['data_table'] = np.array(data['data_table']).astype(float).T
+    data['electron_counts'] = np.array(data['electron_counts']).astype(float)
+
+    # Build a dict_for_calculation which keys are 'PhotonEnergy' and orbitals like '1s1/2', '2s1/2', '2p1/2', '2p3/2', '3s1/2', '3p1/2', '3p3/2'...
+    # and values are connected cross sections and number of electrons of each orbital
+    # This is for calculating the max cross sections of 's', 'p', 'd', 'f' orbitals
+    new_lenth = len(data['electron_counts'])
+    cross_sections_by_orbitals = data['data_table'][-new_lenth:]
+    orbital_headers = data['headers'][-new_lenth:]
+
+    cross_sections_and_electron_numbers = np.concatenate(
+        (cross_sections_by_orbitals.T, [data['electron_counts']]), axis=0).T
+    dict_for_calculation = dict(
+        zip(orbital_headers, cross_sections_and_electron_numbers))
+    energy_array = data['data_table'][0]
+    dict_for_calculation['PhotonEnergy'] = energy_array
+
+    # match the import energy
+    index = np.where(dict_for_calculation['PhotonEnergy'] == energy)[0]
+
+    # build result dict
+    orbitals_cross_sections_dict = {}
+
+    # result for s orbital
+    s_cross_sections = np.array([dict_for_calculation[key]
+                                for key in dict_for_calculation if 's' in key]).T[index]
```

Comment:
The indentation is slightly off here; `for` should align with `dict` because it is inside the `[`. I think a PEP8 linter will catch it?

There is a convenient way of avoiding lots of `my_dict[key]` when iterating over dictionaries, using the `items()` method. An equivalent way of writing this list comprehension would be:
```
[value for key, value in dict_for_calculation.items() if 's' in key]
```
As `for key, value in some_dict.items()` is a common pattern in Python it is easy to read. What you have already is ok in this case, but it's worth knowing this trick for when things get messier!

All the transpose `.T` operations are making it a little bit difficult to keep track of what is being indexed at a given time. Maybe it would help if `index` was named `energy_index`?